src/Data/RFC5322/Internal.hs

```diff
@@ -22,24 +46,196 @@ module Data.RFC5322.Internal
 
   ) where
 
-import Control.Applicative ((<|>), Alternative, liftA2, many)
+import Prelude hiding (takeWhile)
+import Control.Applicative ((<|>), Alternative, liftA2, many, optional)
 import Control.Monad (void)
-import Data.Attoparsec.ByteString as A
-import Data.Attoparsec.Internal as A
+import Control.Lens.Cons (Cons, cons)
+import qualified Data.Attoparsec.ByteString as A
+import qualified Data.Attoparsec.Internal as A
 import qualified Data.Attoparsec.Internal.Types as AT
+import qualified Data.Attoparsec.Text as AText
 import qualified Data.ByteString as B
+import Data.ByteString.Internal (c2w, w2c)
```

Comment:
Aaah.. interesting. TIL

Reply:
@romanofski Use with care; `c2w` simply truncates the `Char` to 8 bits.

Reply:
So does that mean if we have characters encoded with more than 8bits (e.g. utf-16, utf-32) we'd get into trouble? /me hopes he's not mixing something up...

Reply:
`c2w :: Char -> Word8`.  So it doesn't deal with encodings at all, but unicode codepoints.

So if you give it a char > 256, it will lose information. And if you give it a char between 128 and 255, it will not lose information, but the resulting byte will not be a valid utf-8 encoding of that char.

So we have to be careful to only use it with chars (codepoints) < 128.  ATM we only use it directly in two places: `fromChar` and `singleton`.  All the cases where we use these functions deal with chars < 128, so we're OK.